test/loopback-dl.test.js

```diff
@@ -837,6 +837,81 @@ describe('Models attached to a dataSource', function() {
 
   });
 
+  context('replaceAttributes', function() {
+    it('should replace with new data', function(done) {
+      Post.create({title: 'a', content: 'AAA', comments: ['Comment1']},
+      function(err, post) {
+        if (err) return done(err);
+        post.replaceAttributes({title: 'b'}, function(err, p) {
+          if (err) return done(err);
+          p.id.should.equal(post.id);
+          p.should.not.have.property('_id');
+          p.title.should.equal('b');
+          p.should.not.have.property('content', undefined);
+          p.should.not.have.property('comments', undefined);
+          Post.findById(post.id, function(err, p) {
+            if (err) return done(err);
+            p.id.should.equal(post.id);
+            p.should.not.have.property('_id');
+            p.title.should.equal('b');
+            p.should.have.property('content', undefined);
+            p.should.have.property('comments', undefined);
+            done();
+          });
+        });
+      });
+    });
+  });
+
+  context('replaceOrCreate', function() {
+    it('should replace with new data', function(done) {
+      Post.create({title: 'a', content: 'AAA', comments: ['Comment1']},
+        function(err, post) {
+          if (err) return done(err);
+          post = post.toObject();
+          delete post.comments;
+          delete post.content;
+          post.title = 'b';
+          Post.replaceOrCreate(post, function(err, p) {
+            if (err) return done(err);
+            p.id.should.equal(post.id);
+            p.should.not.have.property('_id');
+            p.title.should.equal('b');
+            p.should.not.have.property(p.content);
+            p.should.not.have.property(p.comments);
+            Post.findById(post.id, function(err, p) {
+              if (err) return done(err);
+              p.id.should.equal(post.id);
+              p.should.not.have.property('_id');
+              p.title.should.equal('b');
+              p.should.have.property('content', undefined);
+              p.should.have.property('comments', undefined);
+              done();
+            });
+          });
+        });
+    });
+
+    it('should create a new instance if it does not exist', function(done) {
+      var post = {id: 123, title: 'a', content: 'AAA'};
+      Post.replaceOrCreate(post, function(err, p) {
+        if (err) return done(err);
+        p.id.should.equal(post.id);
+        p.should.not.have.property('_id');
+        p.title.should.equal(post.title);
+        p.content.should.equal(post.content);
+        Post.findById(p.id, function(err, p) {
+          if (err) return done(err);
+          p.id.should.equal(post.id);
+          p.should.not.have.property('_id');
+          p.title.should.equal(post.title);
+          p.content.should.equal(post.content);
+          done();
+        });
+      });
+    });
+  });
+  
```

Comment:
If we have enough tests in `manipulation.test.js` then there is no need to add any tests to `loopback-dl.test.js`.